snap/info.go

```diff
@@ -254,12 +274,16 @@ func ReadInfo(name string, si *SideInfo) (*Info, error) {
 		return nil, err
 	}
 
-	info, err := InfoFromSnapYaml(meta)
+	return infoFromSnapYamlWithSideInfo(meta, si, false)
```

Comment:
it's because the Validate call was moved around, will move again and this will not be an issue